core/course_controller/CourseController.py

```diff
@@ -127,15 +164,31 @@ def search_courses(course_search: str, semester_name: str | None = None) -> List
         ]
     
     @staticmethod
-    def delete_course(course_id: int) -> None:
+    def delete_course(course_id: int, semester_name:str) -> None:
         """
         Pre-conditions: Course id is a valid value matching a record in courses.
         Post-conditions: Removes a record from Course with matching course_id.
         Side-effects: removes matching record from Course table and any objects with foreign key references to it.
         """
         try:
-            course = Course.objects.get(id=course_id)
+            course = Course.objects.get(id=course_id, semester__semester_name=semester_name)
             course.delete()
         except Course.DoesNotExist:
-            raise ValueError("Course with the given code does not exist.")
+            raise ValueError(f"Course {course_id} does not exist.")
 
+    @staticmethod
+    def validate_semester(semester_name: str) -> bool:
```

Comment:
I had said we don't need this method. If you think we do then lets discuss it as a group in the discord